tests/Fixture/FileFixture.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace IterTools\Tests\Fixture;
+
+class FileFixture
+{
+    /**
+     * @param array<string> $lines
+     * @return resource
+     */
+    public static function createFromLines(array $lines)
+    {
+        return self::createFromString(implode("\n", $lines));
+    }
+
+    /**
+     * @param string $string
+     * @return resource
+     */
+    public static function createFromString(string $string)
+    {
+        $file = tmpfile();
+        fwrite($file, $string);
+        fseek($file, 0);
```

Comment:
Suggestion: Take a look at this project: https://github.com/bovigo/vfsStream. I haven't looked too deeply at it, but it might allow you to write unit tests involving files without an infrastructure dependency on an actual file system.

Reply:
Thank you! I'll try.

Reply:
Hi @markrogoyski [I've used](https://github.com/markrogoyski/itertools-php/pull/19/commits/f2f34f7eedecb7d27d6fb982cf4358e8d930fd60) the package you recommended. Now `FileFixture` do not use real filesystem.

Reply:
Nice. Thanks for investigating that alternative as a way to interact with files without involving the actual file system. I'm glad it worked out. It will make for more reliable unit testing.